### ISSUE TITLE
Tests are quietly skipped when @Testdir field is static final

### ISSUE THREAD

**Issue by DomGarguilo** [junit-team/junit5#2837]
We had some tests with a static final `@Tempdir` field which were silently being skipped, going unnoticed for some time (due to the large number of tests in the project, https://github.com/apache/accumulo). I now understand that its best to keep `@Tempdir` fields static and non-final.

## Steps to reproduce

Here is an example test that illustrates the issue:
```java
public class TempdirTest {
    
    @TempDir
    private static final File tempDir = new File(System.getProperty("user.dir"),"TempDir/");

    @Test
    public void testTempDir() {
        assertTrue(tempDir.isDirectory());
    }
}
```

When this test is run it gives the output: 
`Tests run: 0, Failures: 0, Errors: 0, Skipped: 0`
It was just by chance that we noticed the tests were not being run and looked into it more.
## Context

 - For context, here is the PR that addresses the affected areas in our code: https://github.com/apache/accumulo/pull/2528 
 - Used versions (Jupiter/Vintage/Platform): 
   - junit-jupiter-api 5.8.2
 - Build Tool/IDE: 
   - Apache Maven 3.6.3
   - Java version: 11.0.13, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-amd64
   - OS name: "linux", version: "5.15.23-76051523-generic", arch: "amd64", family: "unix"

## Deliverables

- [ ] Make the tests fail when this happens

**Comment by sbrannen** [comment]
Is the combination of `static final` necessary for tests to be silently skipped?

Or is it just because the fields are `final`?

**Comment by DomGarguilo** [comment]
> Is the combination of `static final` necessary for tests to be silently skipped?
> 
> Or is it just because the fields are `final`?

Running a quick test leads me to believe it is the combination of `static final` that causes the tests to be silently skipped.

**Comment by DomGarguilo** [comment]
> Out of curiosity, why would you ever initialize a `@TempDir` field, since the field will be set by JUnit?

I do not see a reason that one would need to do this. I originally did not know that the field is set by JUnit so I manually initialized them which lead me to find this.

**Comment by sbrannen** [comment]
Thanks for providing feedback.

We'll look into it.

**Comment by sbrannen** [comment]
FWIW, there should be an exception thrown for the container (the test class in this scenario).

In Eclipse, the test class fails as a result.

So I suppose it may just be a bug with Maven Surefire in that it does not report the failure at the test class level.

Which version of Maven Surefire are you using?

**Comment by DomGarguilo** [comment]
> Which version of Maven Surefire are you using?

3.0.0-M4

**Comment by sbrannen** [comment]
> 3.0.0-M4

What happens if you upgrade to Maven Surefire 3.0.0-M5?

Also, what happens is you execute the same tests in your IDE (and which IDE)?

**Comment by DomGarguilo** [comment]
> What happens if you upgrade to Maven Surefire 3.0.0-M5?

I don't think we can update the project to 3.0.0-M5 right now but I created a separate project that just contains the example test from above and does not even have Maven Surefire as a dependency. Running the test there produces the same results.

> Also, what happens is you execute the same tests in your IDE (and which IDE)?

Running the tests in IntelliJ IDEA produce the same results as well, the tests are skipped.

**Comment by sbrannen** [comment]
AFAIK, this was fixed in Maven Surefire 3.0.0-M5.

Resources:

- https://issues.apache.org/jira/browse/SUREFIRE-1688
- https://issues.apache.org/jira/browse/SUREFIRE-1727
- https://issues.apache.org/jira/browse/SUREFIRE-1739
- https://issues.apache.org/jira/browse/SUREFIRE-1741
- https://github.com/junit-team/junit5/issues/2386#issuecomment-675676298

I am therefore closing this as a duplicate of #2386.

If you are experiencing similar issues with IntelliJ IDEA, please report the bug to IntelliJ.

Thanks

**Comment by sbrannen** [comment]
To improve the diagnostics in such use cases, we have created:

- #2839
